clang/test/SemaSYCL/kernelname-enum.cpp

```diff
@@ -0,0 +1,79 @@
+// RUN: %clang_cc1 -I %S/Inputs -fsycl -fsycl-is-device -fsycl-int-header=%t.h -fsyntax-only -verify %s
```

Comment:
Did I get it right that diagnostic is not emitted unless integration header is not requested?

Reply:
Yes. All the checks for kernel name type currently seems to be happening when the integration header is being generated.

Reply:
Okay, thanks.